packages/core/src/components/graphs/gauge.ts

```diff
@@ -0,0 +1,316 @@
+
+		// Add the numbers at the center
+		const numbersG = DOMUtils.appendOrSelect(svg, "g.gauge-numbers").attr(
+			"transform",
+			`translate(0, ${numbersYPosition})`
+		);
+
+		// Add the big number
+		const valueNumberG = DOMUtils.appendOrSelect(
+			numbersG,
+			"g.gauge-value-number"
+		).attr("transform", `translate(-10, 0)`); // Optical centering for the presence of the smaller % symbol
+
+		const numberFormatter = Tools.getProperty(options, "gauge", "numberFormatter");
+		const valueNumber = valueNumberG.selectAll("text.gauge-value-number")
+			.data([value]);
+
+		valueNumber
+			.enter()
+			.append("text")
+			.attr("class", "gauge-value-number")
+			.merge(valueNumber)
+			.style("font-size", `${valueFontSize}px`)
+			.attr("text-anchor", "middle")
+			.text(d => numberFormatter(d));
+
+		const {
+			width: valueNumberWidth
+		} = DOMUtils.getSVGElementSize(DOMUtils.appendOrSelect(svg, "text.gauge-value-number"), { useBBox: true });
+
+		DOMUtils.appendOrSelect(valueNumberG, "text.gauge-value-symbol")
+			.style("font-size", `${valueFontSize / 2}px`)
+			.attr("x", valueNumberWidth / 2)
+			.text("%");
+
+		// Add the smaller number of the delta
+		const deltaGroup = DOMUtils.appendOrSelect(
+			numbersG,
+			"g.gauge-delta"
+		).attr(
+			"transform",
+			`translate(0, ${deltaFontSize + distanceBetweenNumbers})`
+		);
+
+		const deltaNumber = deltaGroup.selectAll("text.gauge-delta-number")
+			.data(delta != null ? [delta] : []);
+
+		deltaNumber
+			.enter()
+			.append("text")
+			.merge(deltaNumber)
+			.attr("class", "gauge-delta-number")
+			.attr("text-anchor", "middle")
+			.style("font-size", `${deltaFontSize}px`)
+			.text(d => `${numberFormatter(d)}%`);
+
+		// Add the caret for the delta number
+		const {
+			width: deltaNumberWidth
+		} = DOMUtils.getSVGElementSize(DOMUtils.appendOrSelect(svg, "g.gauge-delta"), { useBBox: true });
+
+		const deltaArrow = deltaGroup.selectAll("svg.gauge-delta-arrow")
+			.data(delta != null ? [delta] : []);
+
+		deltaArrow
+			.enter()
+			.append("svg")
+			.merge(deltaArrow)
+			.attr("class", "gauge-delta-arrow")
+			.attr("x", -arrowSize - deltaNumberWidth / 2)
+			.attr("y", -arrowSize / 2 - deltaFontSize * 0.35)
+			.attr("width", arrowSize)
+			.attr("height", arrowSize)
+			.attr("viewBox", `0 0 16 16`);
+
+		// Needed to correctly size SVG in Firefox
+		DOMUtils.appendOrSelect(deltaArrow, "rect.gauge-delta-arrow-backdrop")
+			.attr("width", `16`)
+			.attr("height", `16`)
+			.attr("fill", "none");
+
+		// Draw the arrow with status
+		const status = Tools.getProperty(options, "gauge", "status");
+		DOMUtils.appendOrSelect(deltaArrow, "polygon.gauge-delta-arrow-polygon")
+			.classed(`status--${status}`, status != null)
+			.attr("fill", () => status == null ? "currentColor" : null)
+			.attr("points", self.getArrow(delta));
+
+		deltaArrow.exit().remove();
+		deltaNumber.exit().remove();
+		arcValue.exit().remove();
+
+		// Add event listeners
+		this.addEventListeners();
+	}
```

Comment:
some of the different pieces of this render function could be taken out into separate functions

Reply:
make a judgement call